text/primitive-projections.md

```diff
@@ -0,0 +1,70 @@
+  - Advantages: the two representations are fully equivalent (up to the compact representation and the loss of the exact form of parameters)
+  - Drawbacks: the code to identify `Const` and `Proj` might be considered intrusive (but at least, it is alreay there, using `Projection.constant` and `expand_projection` machinery)
+
+*Design subchoice 2.1*: When the syntax `f params t` is used, does it bind automatically to the compact representation (as it is done now), and thus reprinted as `t.(f params)`?
+  - Advantages: compatibility with the current usages
+  - Drawbacks: it does some transformation under the carpet
+
+*Design subchoice 2.2*: Any application of `Const f` to enough arguments is automatically turned by `mkApp` into a `Proj` (this assumes that any constant `f` contains an information on its projection status)? In particular, `Const` is used only for non-applied or not-enough-applied projection.
+  - Advantages: a simple model; no need to identify `Proj (f,x)` and `App (Const f, params, x)` in conversion
+  - Drawbacks: maybe a few incompatibilities due to the dropping of arguments?
+
+Related: [#14084](https://github.com/coq/coq/pull/14084) (`match goal` involving primitive projections).
+
+## Issue 2: the relative status of "folded" vs "unfolded" `Proj` node
+
+*Design choice*: The "folded" vs "unfolded" boolean is dropped. Unfolding a `Proj` node is a no-op. In scenario 2.1 of issue 1, unfolding a `Const denoting a projection reveals a `Proj` node. In scenario 2.2, it is a no-op.
+
+See associated [Zulip discussion](https://coq.zulipchat.com/#narrow/stream/237656-Coq-devs.20.26.20plugin.20devs/topic/Primitive.20Projection.20mode).
+
+## Issue 3: the status of `match` and destructing `let` in "primitive" record types
+ 
+See e.g. [#13913](https://github.com/coq/coq/pull/13913).
+
+*Design choice 1*: keep `match` nodes as they are in primitive record types; they will behave correctly for reduction (that is like their expansion into projections); to deal with them in the conversion, it is enough to extend the conversion with eta-expansions of `match` into projections.
+
+## Issue 4: The status of the `t.(f args)` notation as an alternative notation for any kind of application.
+
+See e.g. [#11332](https://github.com/coq/coq/pull/11332).
+
+*Design choice 1*: Accept the notation `t.(f args)` as a shorthand for `f args t` even when `f` is not a projection even if this is lost at printing time.
+  - Advantages: nothing more to do
+  - Drawbacks: no printing
+
+*Design choice 2*: Give the ability to declare a table of "methods" supporting the `t.(f args)` notation. Only such registered method can use the notation. The notation is used for printing too. Internally, it is an `App` node.
+
+*Orthogonal design choice*: same as design choices 1 and 2 but with a different notation, e.g `t.{f args}`.
+
+## Issue 5: The representation of projections in "positive" types. Can they take benefit of a compact representation?
```

Comment:
What does positive/negative mean precisely?

Reply:
I guess the constructor vs destructor variants of coinductive types. I'm not really sure primitive projections make sense for positive coinductive types?

Reply:
Projections make sense for positive records as macro-definitions, as it has been for a long time in Coq. So, the answer depends on which component of "primitive" projections you are talking about. If you are talking about the compact representation, it makes sense to represent the defined projections of positive record types using a dedicated compact `Proj` node, even if thought as macros, i.e. not as a primitive ingredient of positive record types.